### PR TITLE
Add 'squashfs' mount type to fuse-wake

### PR DESCRIPTION
**Pull request by mmjconolly**
Adds a `squashfs` mount operation to fuse-wake

The input json to fuse-wake can have a entry that looks like
```
"mounts": [
  { "type": "squashfs", "source": "your-os-image.squashfs", "destination": "/mnt" }
]
```

Notes:
* This `fork()`s out to `squashfuse`
* `prctl` is used to kill `squashfuse` when `fuse-wake` terminates
* We poll to find out when the mount is ready before continuing\
* Requires `isolate/user` resource

### CODE REVIEW

File: fuse/fuse.cpp [hunk 1]
```diff
@@ -175,6 +181,69 @@ static bool mount_tmpfs(const std::string& destination) {
...
+			std::cerr << "squashfuse prctl: " << strerror(errno) << std::endl;
+			exit(1);
+		}
+		execlp("squashfuse", "squashfuse", "-f", source.c_str(), mountpoint.c_str(), NULL);
+		std::cerr << "execle squashfuse: " << strerror(errno) << std::endl;
+		exit(1);
+	}
+
+	// Wait up to 1 second for the mount to exist before we continue.
+	for(int i = 1; i < 100; i++) {
```
Comment (mmjconolly) [inline fuse/fuse.cpp:236]: exp backoff

File: fuse/fuse.cpp [hunk 2]
```diff
@@ -175,6 +181,69 @@ static bool mount_tmpfs(const std::string& destination) {
...
+		if (is_fuse_filesystem(mountpoint) && is_squashfuse_mount(mountpoint))
```
Comment (mmjconolly) [inline fuse/fuse.cpp:237]: Use stat's st_dev

File: fuse/fuse.cpp [hunk 3]
```diff
@@ -38,6 +38,8 @@
 // For unshare and bind mount
 #include <sched.h>
 #include <sys/mount.h>
+#include <sys/prctl.h>
+#include <sys/types.h>
```
Comment (mmjconolly) [inline fuse/fuse.cpp:42]: It looks like some of the build targets (like debian) want `#define _BSD_SOURCE` for using major/minor with sys/types.h.
I'm suprised about this, as on a ubuntu 16.04 it doesn't require it

Comment (mmjconolly) [reply fuse/fuse.cpp:43]: an musl on alpine wants something else too

**Review by terpstra — APPROVED** [review]
This looks great!

**Pull request merged**